apps/ledger-live-mobile/src/components/RequiresCameraPermissions/useCameraPermissions.ts

```diff
@@ -0,0 +1,59 @@
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import { Camera } from "expo-camera";
+import { AppState, Linking } from "react-native";
+import { useIsMounted } from "../../helpers/useIsMounted";
+
+export default function useCameraPermissions() {
+  const [permission, requestPermission, checkPermission] =
+    Camera.useCameraPermissions();
+  const [firstAutomaticRequestCompleted, setFirstAutomaticRequestCompleted] =
+    useState<boolean>(false);
+  const isMounted = useIsMounted();
+
+  useEffect(() => {
+    requestPermission().then(() => setFirstAutomaticRequestCompleted(true));
+    // only run this once on mount
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const appState = useRef(AppState.currentState);
+  const shouldCheckPermissionOnNextResume = useRef(false);
+  useEffect(() => {
+    const subscription = AppState.addEventListener("change", nextAppState => {
+      if (
+        appState.current.match(/inactive|background/) &&
```

Comment:
Interested for me on Bluetooth and location: is it needed to check that the previous app state was `inactive` or `background` ? The `addEventListener` reacts on `change`: does it trigger `active` several times in a row ?

Reply:
I don't know exactly, I think there are ways it can happen on Android. This is from the [react-native documentation template](https://reactnative.dev/docs/appstate#basic-usage).